ledger/recovering-indexer-integration-tests/src/test/suite/scala/com/digitalasset/platform/indexer/RecoveringIndexerIntegrationSpec.scala

```diff
@@ -348,4 +327,65 @@ object RecoveringIndexerIntegrationSpec {
 
     private class StateUpdatesFailedException extends RuntimeException("State updates failed.")
   }
+
+  class PartyOnlyQueueWriteService(
+      ledgerId: String,
+      participantId: Ref.ParticipantId,
+  )(implicit materializer: Materializer)
+      extends WritePartyService
+      with ReadService {
+
+    private val (queue, source) = Source.queue[(Offset, Update)](10).preMaterialize()
+    private val offset = new AtomicLong(0)
+    // required for the indexer to resubscribe to the update source
+    private val broadcastSource = source.runWith(BroadcastHub.sink)
+    private val stateSeenSoFar = scala.collection.mutable.Buffer.empty[(Offset, Update)]
+
+    override def ledgerInitialConditions(): Source[LedgerInitialConditions, NotUsed] =
+      Source.repeat(
+        LedgerInitialConditions(
+          ledgerId,
+          LedgerReader.DefaultConfiguration,
+          Time.Timestamp.Epoch,
+        )
+      )
+    override def stateUpdates(beginAfter: Option[Offset])(implicit
```

Comment:
Nit: Adding a new line between `def`s may improve readability. Making it shorter is also fine.